app/services/puppet_fact_parser.rb

```diff
@@ -140,6 +140,19 @@ def cores
     facts.dig('processors', 'count') || facts['processorcount']
   end
 
+  def disks_total
+    disks      = facts['disks']
+    disks_size = 0
+
+    if disks.any?
+      disks.values.each do |disk_attrs|
+        disks_size += (disk_attrs['size_bytes'].to_f rescue 0)
```

Comment:
`rescue`s in ruby are quite expensive, and fact parsing is a very performance-heavy area, so it would be better to handle differently. For example,
```suggestion
        disks_size += disk_attrs&.fetch('size_bytes', 0).to_f
```
Do we actually need this in number a float?

Reply:
If they're bytes, it doesn't make sense to use a float since there is no unit smaller than a byte. A short and safe way to write it would be:
```ruby
disks_size = disks.sum { |disk| disk&.fetch('size_bytes', 0) }
```